src/lib/eventHandler/pasteHandler.ts

```diff
@@ -0,0 +1,131 @@
+import { decodeResponse, isImageResponse } from '../http'
+import { isDesktopOrMobileApp } from '../platform'
+
+const handlePasteText = (
+  doc: CodeMirror.Doc,
+  change: CodeMirror.EditorChange,
+  pastedText: string
+) => {
+  doc.replaceRange(pastedText, change.from, {
+    line: change.from.line,
+    ch: change.from.ch + pastedText.length,
+  })
+}
+
+const handlePasteUrl = async (doc: CodeMirror.Doc, change: CodeMirror.EditorChange, pastedText: string) => {
+  const taggedUrl = `<${pastedText}>`
+  const urlToFetch = pastedText
+
+  doc.replaceRange(taggedUrl, change.from, {
+    line: change.from.line,
+    ch: change.from.ch + pastedText.length,
+  })
+
+  const response = await fetch(urlToFetch, { method: 'get' })
+  let replacement = pastedText
+  if (!isImageResponse(response)) {
+    replacement = await mapNormalResponse(response, urlToFetch)
+  }
+  doc.replaceRange(replacement, change.from, {
+    line: change.from.line,
+    ch: change.from.ch + pastedText.length + 2,
+  })
+}
+
+const mapNormalResponse = async (response: Response, pastedTxt: string) => {
+  const body = await decodeResponse(response)
+  try {
+    const escapePipe = (str: string) => str.replace('|', '\\|')
+    const parsedBody = new window.DOMParser().parseFromString(body, 'text/html')
+    return `[${escapePipe(parsedBody.title)}](${pastedTxt})`
+  } catch (e) {
+    console.log(e)
+  }
+  return ''
+}
+
+export const codeMirrorPasteHandler = (
+  editor: CodeMirror.Editor,
+  change: CodeMirror.EditorChange,
+  enableAutoFetchWebPageTitle: boolean
+) => {
+  if (!isDesktopOrMobileApp()) return
+  if (!enableAutoFetchWebPageTitle) return
+  const editorDoc = editor.getDoc()
+
+  const isUrl = (str: string) => {
+    return /(?:^\w+:|^)\/\/(?:[^\s\.]+\.\S{2}|localhost[\:?\d]*)/.test(str)
+  }
+
+  const isInLinkTag = (doc: CodeMirror.Doc) => {
+    const startCursor = doc.getCursor('start')
+    const prevChar = doc.getRange(
+      {
+        line: startCursor.line,
+        ch: startCursor.ch - 2,
+      },
+      {
+        line: startCursor.line,
+        ch: startCursor.ch,
+      }
+    )
+    const endCursor = doc.getCursor('end')
+    const nextChar = doc.getRange(
+      {
+        line: endCursor.line,
+        ch: endCursor.ch,
+      },
+      {
+        line: endCursor.line,
+        ch: endCursor.ch + 1,
+      }
+    )
+    return prevChar === '](' && nextChar === ')'
```

Comment:
Maybe change variable name to `prevChars`, as it holds 2 characters?

Reply:
I agree. I'll change prevChar to prevChars.